frontend/src/store/mutations.js

```diff
@@ -14,6 +15,9 @@ export const mutations = {
   [SET_LOGIN_ERROR](state, loginError) {
     Object.assign(state, { loginError });
   },
+  [ADD_ROUTE](state, route) {
+    Object.assign(state, { routes: [...state.routes, Object.assign({}, route)] });
```

Comment:
Is the second `Object.assign` required?

Reply:
sorry I meant 
`Object.assign(state, { routes: [...state.routes, route] });`
I am not sure if that passes by value or reference, but then you could do
`Object.assign(state, { routes: [...state.routes, { ...route } ] });`